rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/moderationrequest/Sw360ModerationRequestService.java

```diff
@@ -193,10 +240,87 @@ public Map<PaginationData, List<ModerationRequest>> getRequestsByState(User sw36
+                        request.getComponentDeletions(), reviewer);
+            }
+        }
+            break;
+
+        case PROJECT: {
+            if (request.isRequestDocumentDelete()) {
+                sw360ProjectClient.deleteProject(request.getDocumentId(), reviewer);
+            } else {
+                sw360ProjectClient.updateProjectFromModerationRequest(request.getProjectAdditions(),
+                        request.getProjectDeletions(), reviewer);
+            }
+        }
+            break;
+        case RELEASE: {
+            if (request.isRequestDocumentDelete()) {
+                RequestStatus deleteStatus = sw360ComponentClient.deleteRelease(request.getDocumentId(), reviewer);
+                if (deleteStatus.equals(RequestStatus.SUCCESS)) {
+                    SW360Utils.removeReleaseVulnerabilityRelation(request.getDocumentId(), userFromRequest);
+                }
+            } else {
+                sw360ComponentClient.updateReleaseFromModerationRequest(request.getReleaseAdditions(),
+                        request.getReleaseDeletions(), reviewer);
+            }
+        }
+            break;
+        case LICENSE: {
+            sw360LicenseClient.updateLicenseFromModerationRequest(request.getLicenseAdditions(),
+                    request.getLicenseDeletions(), reviewer, userFromRequest);
+        }
+            break;
+        case SPDX_DOCUMENT: {
+            if (request.isRequestDocumentDelete()) {
+                sw360SPDXClient.deleteSPDXDocument(request.getDocumentId(), reviewer);
+            } else {
+                sw360SPDXClient.updateSPDXDocumentFromModerationRequest(request.getSPDXDocumentAdditions(),
+                        request.getSPDXDocumentDeletions(), reviewer);
+            }
+        }
+            break;
+        case SPDX_DOCUMENT_CREATION_INFO: {
+            if (request.isRequestDocumentDelete()) {
+                documentCreationInfoClient.deleteDocumentCreationInformation(request.getDocumentId(), reviewer);
+            } else {
+                documentCreationInfoClient.updateDocumentCreationInfomationFromModerationRequest(
+                        request.getDocumentCreationInfoAdditions(), request.getDocumentCreationInfoDeletions(),
+                        reviewer);
+            }
+        }
+            break;
+        case SPDX_PACKAGE_INFO: {
+            if (request.isRequestDocumentDelete()) {
+                packageInfoClient.deletePackageInformation(request.getDocumentId(), reviewer);
+            } else {
+                packageInfoClient.updatePackageInfomationFromModerationRequest(request.getPackageInfoAdditions(),
+                        request.getPackageInfoDeletions(), reviewer);
+            }
+        }
+            break;
+        }
         getThriftModerationClient().acceptRequest(request, moderatorComment, reviewer.getEmail());
         return ModerationState.APPROVED;
```

Comment:
The status here should be dependent on response of action performed like deletion or updation? Right now, the endpoint always accept the request and return APPROVED status.

Reply:
comment Addressed.